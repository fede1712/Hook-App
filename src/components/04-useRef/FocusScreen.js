import React, { useRef } from 'react'
import '../02-useEffect/SimpleForm.css'

export const FocusScreen = () => {

    const inputRef = useRef()
    const handleClik = () => {
        inputRef.current.select()
    }

    return (
        <div>
            <h1>FocusScreen</h1>
            <hr />

            <input
                ref={inputRef}
                className='form-control'
                placeholder='Your Name'
            />
            <button
                className='btn btn-outline-primary mt-5'
                onClick={handleClik}
            >
                Focus

            </button>

        </div>
    )
}
