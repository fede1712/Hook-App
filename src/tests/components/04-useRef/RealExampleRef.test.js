import React from 'react'
import { shallow } from 'enzyme'
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef"


describe('Tests in <RealExample />', () => {

    const wrapper = shallow(<RealExampleRef />)

    test('should render <RealExample /> correctly', () => {

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false)

    })

    test('should render the component <MultipleCustomHooks />', () => {

        wrapper.find('button').simulate('click')

        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true)

    })

})
