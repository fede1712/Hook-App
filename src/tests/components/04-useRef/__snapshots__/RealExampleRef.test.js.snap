// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <RealExample /> should render <RealExample /> correctly 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-outline-secondary mt-5"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</div>
`;
